### PR TITLE
fix(type): only return Promise with delay

### PR DESCRIPTION
**Pull request by nickmccurdy**
**What**: The TypeScript type for the `type` function will now only return a `Promise` when given a `delay` option.

**Why**: #480

**How**: Updated type declaration and added basic type tests.

**Checklist**:

- [ ] Documentation N/A
- [x] Tests
- [x] Typings
- [x] Ready to be merged

### CODE REVIEW

**Review by kentcdodds — APPROVED** [review]
Nice! This is great 👍 thank you!

**Comment by kentcdodds** [comment]
Ugh, we had this build issue elsewhere: https://github.com/rollup/plugins/issues/640

File: typings/test.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import userEvent, {TargetElement} from '.'
+
+declare const element: TargetElement
+type NotVoid = string | number | boolean | object | null
+
+userEvent.type(element, 'foo', {delay: 1}) as Promise<void>
+// @ts-expect-error No delay returns void
```
Comment (luistak) [inline typings/test.ts:7]: Awesome [usage of typescript](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments) this was new to me many thanks! ❤️

**Review by luistak — APPROVED** [review]

**Comment by kentcdodds** [comment]
It looks like our testing code coveraged dipped. So we can merge this, but it won't get released until we fix the code coverage level. From the look of it, it's all around `isContentEditable`:


image

image

image

**Comment by nickmccurdy** [comment]
That's strange, the codecov support says 100% coverage. Is that incorrect or only for the files in this PR?

**Comment by kentcdodds** [comment]
Yeah, very odd. I just ran it locally out of curiosity and got that result. But this build succeeded and the library got published so 🤷‍♂️

**Comment by nickmccurdy** [comment]
I just pulled and ran `npm update` and now I have 100% coverage locally.

**Pull request merged**